### ISSUE TITLE
Make shouldNotBeNull return non-null instance

### ISSUE THREAD

**Issue by goreRatzete** [MarkusAmshove/Kluent#90]
Sometimes you want to test whether a nullable instance is actually null. 
Most often I want to run futher tests on this instance afterwards. 

It would be nice if the `shouldNotBeNull` fun returned a non null type when the test passes (and throws if not).

Sample implementation:
```
fun <T : Any> T?.shouldNotBeNull(): T = this ?: throw AssertionError("Expected this to not be null")
```

This allows one to write the following code:
```
val user = userSource.findUser(details.email).shouldNotBeNull()
user.email.shouldNotBeBlank()
articleSource.insertArticle(user, it)
```
Without having to `!!` every user reference.

**Comment by jcornaz** [comment]
Todaly agree, I often encounter this use case too.

**Comment by MarkusAmshove** [comment]
If you want you can review the [change](https://github.com/MarkusAmshove/Kluent/commit/64c395dd405f8048d7cb15d850fbe334303abdf0) before I release it :-)

**Comment by MarkusAmshove** [comment]
This might also be something worth it for almost all other assertions, to return the value on their happy path

**Comment by MarkusAmshove** [comment]
A good candidate would be 

`infix fun Any?.shouldBeInstanceOf(className: Class<*>) = assertTrue("Expected $this to be an instance of $className", className.isInstance(this))`

to

`inline fun <reified T: Any> Any.shouldBeInstanceOf() : T = if(this::class.isInstance(T::class)) this as T else throw AssertionError("Expected $this to be an instance of ${T::class.qualifiedName}")`

which has to be a separate method to not break existing code.

This would allow:

```kt
open class Base {
    fun doSomething() = println("Base")
}

class Child : Base() {
    fun different() = println("Child")
}

fun main(args: Array<String>) {
    val base : Base = Child()
    // base.different() - Doesn't compile, not of type Child
    val child = base.shouldBeInstanceOf<Child>()
    child.different()
}```

**Comment by goreRatzete** [comment]
Exactly, it should help wherever possible.

Why did you put this as `in next release` and `needs migration to 2.0` ?

**Comment by MarkusAmshove** [comment]
`in next release` means that it'll the in the next version that is released (pushed/merged to master, waiting to be released).

Migration to 2.0 is just to keep track of changes that aren't in the 2.0 branch yet. They have to be migrated by hand, because the repository and project structure changes for 2.0.
It makes sure that 2.0 will be on par with the latest 1.x version feature wise ;-)

**Comment by MarkusAmshove** [comment]
I've just released these changes with v1.34
